.changeset/tricky-wolves-collect.md

```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": minor
+---
+
+Added: support to use `--fix` with `checkAgainstRule`
```

Comment:
[suggest] Since this change is related to not only CLI but also the Node.js API, could you change the wording like this? (I've changed the PR title)

```suggestion
Added: support autofix with `checkAgainstRule`
```